src/pages/Admin/Charity/Permissions/useSubmit.ts

```diff
@@ -0,0 +1,63 @@
+import { useFormContext } from "react-hook-form";
+import { ProposalMeta } from "pages/Admin/types";
+import { useAdminResources } from "pages/Admin/Guard";
+import { useErrorContext } from "contexts/ErrorContext";
+import { useGetWallet } from "contexts/WalletContext";
+import CW3 from "contracts/CW3";
+import SettingsController from "contracts/SettingsController";
+import useCosmosTxSender from "hooks/useCosmosTxSender";
+import { isEmpty } from "helpers";
+import { getPayloadDiff } from "helpers/admin";
+import createUpdateEndowmentControllerMsg from "./createUpdateEndowmentControllerMsg";
+import { FormValues } from "./schema";
+
+export default function useSubmit() {
+  const { id, cw3, propMeta, endow_type } = useAdminResources<"charity">();
+  const { handleError } = useErrorContext();
+  const {
+    formState: { isSubmitting },
+    handleSubmit,
+    reset,
+  } = useFormContext<FormValues>();
+  const { wallet } = useGetWallet();
+  const sendTx = useCosmosTxSender();
+
+  async function onSubmit({ initialValues, ...newValues }: FormValues) {
+    try {
+      const diff = getPayloadDiff(initialValues, newValues);
+
+      if (isEmpty(Object.entries(diff))) {
+        return handleError("No changes detected");
+      }
+
+      const settingsController = new SettingsController(wallet);
+      const msg = createUpdateEndowmentControllerMsg(id, diff);
+      const embeddedMsg =
+        settingsController.createEmbeddedUpdateEndowmentControllerMsg(msg);
+
+      const meta: ProposalMeta = { type: "endow_controller" };
+
+      const adminContract = new CW3(wallet, cw3);
+      const proposalMsg = adminContract.createProposalMsg(
```

Comment:
should have a provision where if `wallet.address === EndowmentController.endowment_controller.delegate || wallet.address === endowent.owner`, should just send the msg to `SettingsController` contract

Reply:
Waiting on Andrey to confirm the exact requirements here, see https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/1885#issuecomment-1469530382.

Reply:
As per the most recent comments https://github.com/AngelProtocolFinance/angelprotocol-web-app/issues/1708#issuecomment-1469557854, `wallet.address === EndowmentController.endowment_controller.delegate` is the only relevant check, updating the code.